### PR TITLE
Enable sandbox on webview

### LINKED ISSUE #9611: Sandbox option doesn't work on webview
* Electron version: 1.7.1
* Operating system: Windows

### Expected behavior
Able to enable sandbox on webview

### Actual behavior
Sandbox option doesn't work on webview

### How to reproduce

Create webview with sandbox option set to true 
<webview src="https://www.google.com" webpreferences="sandbox=yes, nodeIntegration=no"/>

webview process will not have sandbox enabled. 

See the command line options set on the webview process

electron.exe" --type=renderer **--no-sandbox** --primordial-pipe-token=19883710511F93F2300EF53454123946 --lang=en-US --app-path="c:\Repos\electron-quick-start" **--node-integration=false --enable-sandbox** --guest-instance-id=1 --enable-blink-features 

I think we should unset --no-sandbox in the if statement @ [IsSandboxed
](https://github.com/electron/electron/blob/8ae36c250c456b7a334c00109ec9f5f01beab0a4/atom/browser/web_contents_preferences.cc#L114).

@tarruda @kevinsawicki if you guys think this suggestion makes sense, i can send out a PR for the fix.

Repros
  $ git clone https://github.com/juturu/electron-quick-start.git -b webview-sandbox
  $ npm install
  $ npm start || electron .

### PR DESCRIPTION
**Pull request by juturu**
Fixes #9611

### CODE REVIEW

**Comment by juturu** [comment]
@tarruda I have tested your patch and it was having --no-sandbox --sandbox on webview with sandbox:true. This is not enabling chrome's default sandbox mechanism in windows like running in untrusted IL. So, this change is not setting --no-sandbox command line when webcontents has sandbox option set.

**Comment by tarruda** [comment]
@juturu I'm not sure I understand the problem, but I just tested https://github.com/tarruda/electron/tree/enable-webview-on-sandbox with the following app https://gist.github.com/tarruda/947fe6509b8c47a6dbf76866b3278f40 and it worked as expected. Can you try it and paste what you see on console?

**Comment by pwnsdx** [comment]
👍

**Comment by juturu** [comment]
Sorry for the delay on this.
@tarruda  your change does work in enabling webview. We need this change for the following purpose
 
With your change, electron with sandbox'ed webview starts in Medium IL
 
image

With this current PR, processes start in low/untrusted IL making them more secure and identical to chrome. 
image

I have created `--enable-mixed-sandbox` option to get this functionality.

**Comment by juturu** [comment]
@tarruda @kevinsawicki can you please take a look at this one?

**Comment by tarruda** [comment]
@juturu TBH I don't understand what this PR is doing, is it supposed to work on top of my webview/sandbox patch? Because if so, I haven't even created a PR for that yet.

Also, did you test the `--enable-mixed-sandbox` on all platforms? Even if this works on windows, I'm not sure other platforms(linux at least) will be able to have this mixed sandbox idea implemented.

**Comment by juturu** [comment]
>is it supposed to work on top of my webview/sandbox patch?

No, this is working without your webview/sandbox patch.

>did you test the --enable-mixed-sandbox on all platforms?

I have tried this on mac and windows. And I see Sandbox option in mac to be turned on with `--enable-mixed-sandbox` from Activity Monitor. And on windows as mentioned above i see webviews started in different ILs. I don't have Linux machine handy. If anyone can try this out that would help. If you see it won't work on Linux machines, we can make this option only for Mac and Windows.

**Comment by juturu** [comment]
@tarruda i tried adding the test you mentioned. Unfortunately `process` object is undefined when `sandbox: true` causing the test validation not possible.

**Comment by kevinsawicki** [comment]
@tarruda are you on-board with these changes and approach overall?

**Comment by tarruda** [comment]
I think this requires more testing, especially if it shows some real usage of the mixed sandbox argument(AFAICT right now it only tests if the argument is passed).

TBH I still have some doubts if it will work correctly on all platforms, so testing is very important.

**Comment by juturu** [comment]
@tarruda as mentioned earlier, i tested this works correctly in windows and mac. Don't have linux machine to test. As this is a experimental feature and i couldn't add more tests, can we merge this and add more tests as other sandbox issues are fixed (especially process object is null in sandbox mode).

**Comment by juturu** [comment]
@tarruda @kevinsawicki gentle reminder on this.

**Comment by kevinsawicki** [comment]
Thanks for this @juturu 👍

**Comment by juturu** [comment]
Thanks @kevinsawicki 🙇

File: docs/api/sandbox-option.md [hunk 1]
```diff
@@ -60,6 +60,8 @@ It is important to note that this option alone won't enable the OS-enforced sand
 `--enable-sandbox` command-line argument must be passed to electron, which will
 force `sandbox: true` for all `BrowserWindow` instances.
 
+To enable OS-enforced sandbox on `BrowserWindow` or `webview` process with `sandbox:true` without causing
+entire app to be in sandbox, `--enable-mixed-sandbox` command-line argument must be passed to electron.
```
Comment (jdalton) [inline docs/api/sandbox-option.md]: There is a note a bit lower that could use a follow-up mention about `--enable-mixed-sandbox`.

> It is not possible to have the OS sandbox active only for some renderers, if
 `--enable-sandbox` is enabled, normal electron windows cannot be created.
